nixos/modules/services/databases/postgresql.nix

```diff
@@ -60,6 +61,8 @@ let
 
   groupAccessAvailable = versionAtLeast postgresql.version "11.0";
 
+  pluginNames = map (plugin: plugin.pname) cfg.plugins;
+  wantsPlugin = plugin: elem plugin plugin pluginNames;
```

Comment:
Yep, that's true. So now we just read the extraPlugins from the package instead.

Reply:
> This would not work when setting `package` to `postgresqlWithPackages ...`, right?

Sorry, but what is `postgresqlWithPackages`? How would you set it?

Reply:
`postgresqlWithPackages` is the function within `pkgs.postgresql` that gets called when doing `postgresql.withPackages`. Basically a wrapper around `buildEnv`:

https://github.com/NixOS/nixpkgs/blob/a6f2dfc2572daeb1f15baa18ec3331e526660dcc/pkgs/servers/sql/postgresql/generic.nix#L347

Long story short, this somehow needs to expose which plugins it contains.

(Yes, one may be able to use `paths` for that, but that's kinda weird)